### ISSUE TITLE
Chain and cache provider should be registered first

### ISSUE THREAD

**Issue by mpoiriert** [geocoder-php/BazingaGeocoderBundle#98]
Following the documentation for the cache

```
bazinga_geocoder:
    providers:
        cache:
            adapter:  acme_cache_adapter
            provider: google_maps
        google_maps: ~
```

And following the flow of the BazingaGeocoderExtension and the AddProvidersPass the cache (or chain provider) will always be registered at the end so they will be completly useless.

The should probably be registered first (cache than chain) or all the adapters should be registered in the order present in the file so the user have complete control over it (probably what was suppose to be the purpose of the chain adapter...)

If the bug is real I could fix it if you want.

**Comment by mpoiriert** [comment]
I must specify that I am only using the GeocoderListener for doctrine so the  Geocoder::using() is not call.

**Comment by Baachi** [comment]
Feel free to open a PR

**Comment by Nyholm** [comment]
This will be fixed in #127 though we can add cache on all providers. 
Im closing this issue, feel free to open it again if you still having issues.
